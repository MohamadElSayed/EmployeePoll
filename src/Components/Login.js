import { useRef } from "react";
// import { useEffect } from "react";
import { handleSetAuthedUser } from "../actions/authedUser";
import { connect } from "react-redux";

const Login = (props) => {
  const usernameRef = useRef();
  const passwordRef = useRef();

  // useEffect(() => {}, []);

  const handleSubmit = (event) => {
    event.preventDefault();
    props.dispatch(
      handleSetAuthedUser(usernameRef.current.value, passwordRef.current.value)
    );
  };

  return (
    <div className="container">
      <form className="justify-content-center" onSubmit={handleSubmit}>
        <div className="row">
          <h3
            className="col"
            style={{ textAlign: "center", paddingTop: "50px" }}
          >
            Employee Polls
          </h3>
        </div>
        <div className="row justify-content-center">
          <img
            src={require("../Images/login.gif")}
            style={{ width: "200px", height: "200px" }}
          />
        </div>

        <div className="Row">
          <p
            className="text-danger"
            style={{
              width: "70%",
              display: "block",
              marginRight: "auto",
              marginLeft: "auto",
            }}
          >
            {props.authedUser && props.authedUser.errorMessage
              ? props.authedUser.errorMessage
              : ""}
          </p>
        </div>
        <div className="row">
          <div
            className="col"
            style={{ textAlign: "center", paddingTop: "10px" }}
          >
            User
          </div>
        </div>

        <div className="row justify-content-center">
          <div className="col" style={{ paddingTop: "10px" }}>
            <input
              type="text"
              name="User"
              className="form-control"
              style={{
                width: "70%",
                display: "block",
                marginRight: "auto",
                marginLeft: "auto",
              }}
              ref={usernameRef}
            />
          </div>
        </div>

        <div className="row">
          <div
            className="col"
            style={{ textAlign: "center", paddingTop: "10px" }}
          >
            Password
          </div>
        </div>

        <div className="row justify-content-center">
          <div className="col" style={{ paddingTop: "10px" }}>
            <input
              type="password"
              name="Password"
              className="form-control"
              style={{
                width: "70%",
                display: "block",
                marginRight: "auto",
                marginLeft: "auto",
              }}
              ref={passwordRef}
            />
          </div>
        </div>

        <br />
        <br />
        <div className="row justify-content-center">
          <div className="col" style={{ paddingTop: "0px" }}>
            <button
              type="submit"
              name="submitLogin"
              className="btn btn-primary "
              style={{
                width: "20%",
                display: "block",
                marginRight: "auto",
                marginLeft: "auto",
              }}
            >
              Submit
            </button>
          </div>
        </div>
      </form>
    </div>
  );
};

export default connect((state) => ({
  authedUser: state.authedUser,
}))(Login);
