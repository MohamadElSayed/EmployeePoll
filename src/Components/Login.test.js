import Login from "./Login";
import * as React from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import reducer from "../reducers";
import { BrowserRouter } from "react-router-dom";
import { render, fireEvent, waitFor } from "@testing-library/react";
import middleware from "../middleware";

const store = createStore(reducer, middleware);

describe("Login Snapshot Tests", () => {
  let component;
  beforeEach(() => {
    component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );
  });

  it("matching the snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  it("firing Events - check that error message is not present initially", async () => {
    expect(component.queryByTestId("errorMessage")).not.toBeInTheDocument();
  });

  it("firing Events - check for invalid username/password", async () => {
    var userNameInput = component.getByTestId("username");
    fireEvent.change(userNameInput, { target: { value: "Mike" } });

    var passwordInput = component.getByTestId("password");
    fireEvent.change(passwordInput, { target: { value: "password123" } });

    var submitButton = component.getByTestId("submit-button");
    fireEvent.click(submitButton);

    await waitFor(() => {
      expect(component.getByTestId("errorMessage")).toBeInTheDocument();
    });
  });
});
