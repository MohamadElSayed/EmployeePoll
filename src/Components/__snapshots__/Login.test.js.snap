// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Snapshot Tests matching the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <form
          class="justify-content-center"
        >
          <div
            class="row"
          >
            <h3
              class="col"
              style="text-align: center; padding-top: 50px;"
            >
              Employee Polls
            </h3>
          </div>
          <div
            class="row justify-content-center"
          >
            <img
              alt="login"
              src="login.gif"
              style="width: 200px; height: 200px;"
            />
          </div>
          <div
            class="Row"
          />
          <div
            class="row"
          >
            <div
              class="col"
              style="text-align: center; padding-top: 10px;"
            >
              User
            </div>
          </div>
          <div
            class="row justify-content-center"
          >
            <div
              class="col"
              style="padding-top: 10px;"
            >
              <input
                class="form-control"
                data-testid="username"
                name="username"
                style="width: 70%; display: block; margin-right: auto; margin-left: auto;"
                type="text"
              />
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col"
              style="text-align: center; padding-top: 10px;"
            >
              Password
            </div>
          </div>
          <div
            class="row justify-content-center"
          >
            <div
              class="col"
              style="padding-top: 10px;"
            >
              <input
                class="form-control"
                data-testid="password"
                name="password"
                style="width: 70%; display: block; margin-right: auto; margin-left: auto;"
                type="password"
              />
            </div>
          </div>
          <br />
          <br />
          <div
            class="row justify-content-center"
          >
            <div
              class="col"
              style="padding-top: 0px;"
            >
              <button
                class="btn btn-primary "
                data-testid="submit-button"
                name="submitLogin"
                style="width: 20%; display: block; margin-right: auto; margin-left: auto;"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <form
        class="justify-content-center"
      >
        <div
          class="row"
        >
          <h3
            class="col"
            style="text-align: center; padding-top: 50px;"
          >
            Employee Polls
          </h3>
        </div>
        <div
          class="row justify-content-center"
        >
          <img
            alt="login"
            src="login.gif"
            style="width: 200px; height: 200px;"
          />
        </div>
        <div
          class="Row"
        />
        <div
          class="row"
        >
          <div
            class="col"
            style="text-align: center; padding-top: 10px;"
          >
            User
          </div>
        </div>
        <div
          class="row justify-content-center"
        >
          <div
            class="col"
            style="padding-top: 10px;"
          >
            <input
              class="form-control"
              data-testid="username"
              name="username"
              style="width: 70%; display: block; margin-right: auto; margin-left: auto;"
              type="text"
            />
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
            style="text-align: center; padding-top: 10px;"
          >
            Password
          </div>
        </div>
        <div
          class="row justify-content-center"
        >
          <div
            class="col"
            style="padding-top: 10px;"
          >
            <input
              class="form-control"
              data-testid="password"
              name="password"
              style="width: 70%; display: block; margin-right: auto; margin-left: auto;"
              type="password"
            />
          </div>
        </div>
        <br />
        <br />
        <div
          class="row justify-content-center"
        >
          <div
            class="col"
            style="padding-top: 0px;"
          >
            <button
              class="btn btn-primary "
              data-testid="submit-button"
              name="submitLogin"
              style="width: 20%; display: block; margin-right: auto; margin-left: auto;"
              type="submit"
            >
              Submit
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
